UI Enhancements: Sorting Controls, Interactive Heatmap

M1CTIAN opened the pull request:
This PR implements several UI/UX enhancements to make CommitBoard more user-friendly and visually appealing.

✨ Changes Implemented
1. Sorting Functionality
2. Interactive Contribution Heatmap
3. Enhanced Styling & Accessibility

M1CTIAN committed:
Enhance team member data structure and add roles

- Updated team.json to include role and team information for each member.
- Improved dashboard styles for better visual hierarchy and responsiveness.
- Implemented a new server.js file to handle GitHub API requests for fetching contributor stats.
- Created SortingControls component with enhanced sorting and filtering capabilities.
- Added comprehensive styles for sorting controls, including responsive design and accessibility features.

M1CTIAN committed:
refactor: remove filtering options from SortingControls and Dashboard components
style: update dashboard styles to improve layout and remove unnecessary properties

M1CTIAN commented:
@Balaji-V-S

Balaji-V-S reviewed (commented):
Hey this looks great, I'll pull and test on my local env. Meanwhile I've some suggestions in /api/fetch-stats.js and server.js
-> It's not a good practice to log on browser console (in production version)

I'm merge once, I test it out

M1CTIAN commented:
> Hey this looks great, I'll pull and test on my local env. Meanwhile I've some suggestions in /api/fetch-stats.js and server.js -> It's not a good practice to log on browser console (in production version)
> 
> I'm merge once, I test it out

Sorry, I must've forgotten to remove them before pushing them to my fork, I'll ensure no such things happen again

Balaji-V-S reviewed (changes requested):
Arpit just a heads-up your branch wasn’t synced with the latest main, and I’ve moved to TailwindCSS, so some changes may need adjustments

M1CTIAN committed:
integrate Tailwind CSS and refactor styles

- Added Tailwind CSS and its Vite plugin to the project.
- Removed App.css and footer.css, replacing them with Tailwind utility classes in components.
- Created a new globals.css file for global styles and Tailwind imports.
- Updated the footer component to use Tailwind for styling.
- Cleaned up main.jsx by removing unused CSS imports.

M1CTIAN committed:
Update README

M1CTIAN committed:
refactor: update package dependencies and improve dashboard styles

- Moved @tailwindcss/vite and tailwindcss to devDependencies for better organization.
- Enhanced SortingControls component with improved styling for sort labels and indicators.
- Updated Dashboard component styles for better layout and responsiveness, including adjustments to card and header styles.
- Cleaned up dashboard.css by removing unused styles and optimizing existing ones.
- Added base styles for the dashboard in globals.css for consistent theming.
- Configured Vite to use Tailwind CSS correctly.
- Created a new ts.config.js for Tailwind CSS configuration.

Balaji-V-S committed:
Merge branch 'UI-enhancements' of https://github.com/M1CTIAN/CommitBoard into pr#6-UI-enhancements/Arpit-Raj

Balaji-V-S reviewed (commented):
<img width="1888" height="943" alt="image" src="https://github.com/user-attachments/assets/078cb564-7354-46c4-aeb0-f0cd9d8dcb01" />

-> I'm fixing this jaggered alignment issue and the API could be easily accessed with curl with null origin commands. I found this API flaw in my version too, I'm fixing it and merge the changes

->Removing error status from console and screen

M1CTIAN commented:
<img width="1919" height="1049" alt="image" src="https://github.com/user-attachments/assets/bc8d7642-7473-46bb-b551-407464f3d350" />
@Balaji-V-S I dont see any such errors on my dev build. Also I have pulled the changes from main branch and updated the code

Balaji-V-S committed:
Merge branch 'pr#6-UI-enhancements/Arpit-Raj' into UI-enhancements

Balaji-V-S committed:
Merged UI enhancements and updated Sorting styles

Co-authored-by: M1CTIAN <M1CTIAN@users.noreply.github.com>
